### PR TITLE
osd/PG: fix possible overflow on unfound objects

### PR DESCRIPTION
**Pull request by hjwsm1989**
Signed-off-by: huangjun <hjwsm1989@gmail.com>

### CODE REVIEW

File: src/osd/PG.cc [hunk 1]
```diff
@@ -596,11 +596,12 @@ bool PG::MissingLoc::add_source_info(
 void PG::discover_all_missing(map<int, map<spg_t,pg_query_t> > &query_map)
 {
   auto &missing = pg_log.get_missing();
-  assert(have_unfound());
+  uint64_t unfound = get_num_unfound();
+  assert(unfound > 0);
```
Comment (liewegas) [inline src/osd/PG.cc]: the assert isn't needed if it's unsigned

**Comment by badone** [comment]
s/possiable/possible/ in your commit message?

File: src/osd/PG.cc (hunk 1 above)
Comment (xiexingguo) [inline src/osd/PG.cc]: It seems the original logic will confirm we have at least one unfound object here..

Comment (hjwsm1989) [inline src/osd/PG.cc]: yes, there must at least one unfound object.
and dont need to call get_num_unfound() again in next dout()

**Comment by hjwsm1989** [comment]
@badone updated

File: src/osd/PG.cc (hunk 1 above)
Comment (hjwsm1989) [inline src/osd/PG.cc]: @xiexingguo did this change the origin logic?

Comment (xiexingguo) [inline src/osd/PG.cc]: Don't think so. looks good.

**Comment by hjwsm1989** [comment]
retest this please

File: src/osd/PG.cc (hunk 1 above)
Comment (liewegas) [inline src/osd/PG.cc]: nevermind; i was thinking this was >= 0.

**Pull request merged**